source/guide_mailman-3.rst

```diff
@@ -0,0 +1,464 @@
+
+
+Configuration
+=============
+Get a free port
+---------------
+
+We need to find a couple of free ports and bind your application to it.  Since Mailman Core itself exposes a REST interface, Postorius and HyperKitty run as Django applications with their own webserver and we need a LMTP port for local mail forwarding from qmail_ to Mailman, you will need to execute the following code multiple times (directly before you are going to change the respective configurations).
+
+.. include:: includes/generate-port.rst
+
+
+Configure Mailman Core
+----------------------
+
+At first, we need to configure the REST interface of the core component. To create a basic configuration, we need to run ``mailman info`` and get the path of the configuration file. The output should look like this:
+
+:: 
+
+ [isabell@stardust ~]$ mailman info
+ GNU Mailman 3.2.0 (La Villa Strangiato)
+ Python 3.6.7 (default, Dec  5 2018, 15:02:05)
+ [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
+ config file: /home/isabell/var/etc/mailman.cfg
+ db url: sqlite:////home/isabell/var/data/mailman.db
+ devmode: DISABLED
+ REST root url: http://localhost:8001/3.1/
+ REST credentials: restadmin:restpass
+ [isabell@stardust ~]$ 
+
+When the file is created, we need to add and change a couple of options in ``/home/isabell/var/etc/mailman.cfg``. The ``mta`` section contains the configuration related to sending and receiving mails. As we are using qmail_, the incoming MTA has to be set to null. In ``webservice`` section we configure the REST API server. As sometimes the paths where mailman stores its data can depend on the directory from where it is called, we manually define all necessary paths in ``paths.custom`` and force their usage in ``mailman``. A full overview of possible settings can be found in the `mailman docs schema.cfg`_.
+
+.. code :: cfg
+
+ [mta]
+ incoming: mailman.mta.null.NullMTA
+ lmtp_host: localhost
+ smtp_host: stardust.uberspace.de
+ # First free port
+ lmtp_port: 9000
+ smtp_port: 587 
+ smtp_user: forwarder@isabell.uber.space
+ smtp_pass: mailpassword
+
+ [webservice]
+ hostname: localhost
+ # Second free port
+ port: 9001
+ use_https: no
+ admin_user: restadmin
+ admin_pass: restpass
+ api_version: 3.1
+
+ [paths.custom]
+ var_dir: /home/isabell/var
+ bin_dir: /home/isabell/.local/bin
+ log_dir: /home/isabell/var/logs
+ lock_dir: /home/isabell/var/locks
+ data_dir: /home/isabell/var/data
+ cache_dir: /home/isabell/var/cache
+ etc_dir: /home/isabell/var/etc
+ messages_dir: /home/isabell/var/messages
+ archive_dir: /home/isabell/var/archives
+ template_dir: /home/isabell/var/templates
+ pid_file: /home/isabell/var/master.pid
+ lock_file: /home/isabell/var/locks/master.lck
+
+ [mailman]
+ layout: custom
+
+Daemonizing Mailman Core
+------------------------
+
+As we want to make sure that Mailman is started automatically, we need to set it up as a service. Due to mailman not having having the option to always run in foreground, we need some other means of controlling it.
```

Comment:
`mailman start` forks and then executes `.local/bin/master` (yes, it's just called "master") with `--config`. That one doesn't fork and stays in the foreground. Since that executable is publicly available and has a friendly `--help', I guess using it directly would be fine.

The start helper also does the following:

```
os.environ['MAILMAN_VAR_DIR'] = config.VAR_DIR
os.chdir(config.VAR_DIR)
```

so we could adapt the supervisord config to read

```
environment=MAILMAN_VAR_DIR="%(HOME)s/var"
directory=%(HOME)s/var
```

Although looking at [config.py](https://gitlab.com/mailman/mailman/blob/master/src/mailman/config/config.py#L158), where this variable is ultimately used, I am pretty sure we can skip that if all paths in the config are absolute (which they are).

Reply:
Not knowing that much about Python coding best practices, this seems somehow weird 😕 
But great that you found this out, will adapt the supervisord config.

If we set the environment variable and the working directory from supervisor, we probably don't even need to hard-code absolute paths into the config 🎉